qualtran/serialization/sympy.py

```diff
@@ -0,0 +1,222 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Any, Union
+
+import sympy
+
+from qualtran.protos import sympy_pb2
+
+
+def _get_sympy_function_type(expr: sympy.Expr) -> int:
+    """
+    Helper function for serializing a sympy function.
+
+    This method converts a sympy function to its sympy_pb2.Function enum representation.
+    """
+    if isinstance(expr, sympy.core.mul.Mul):
+        return sympy_pb2.Function.Mul
+    elif isinstance(expr, sympy.core.add.Add):
+        return sympy_pb2.Function.Add
+    elif isinstance(expr, sympy.core.power.Pow):
+        return sympy_pb2.Function.Pow
+    elif isinstance(expr, sympy.core.Mod):
+        return sympy_pb2.Function.Mod
+    elif isinstance(expr, sympy.functions.elementary.exponential.log):
+        return sympy_pb2.Function.Log
+    elif isinstance(expr, sympy.functions.elementary.integers.floor):
+        return sympy_pb2.Function.Floor
+    elif isinstance(expr, sympy.functions.elementary.integers.ceiling):
+        return sympy_pb2.Function.Ceiling
+    elif isinstance(expr, sympy.functions.elementary.miscellaneous.Max):
+        return sympy_pb2.Function.Max
+    elif isinstance(expr, sympy.functions.elementary.miscellaneous.Min):
+        return sympy_pb2.Function.Min
+    elif isinstance(expr, sympy.functions.elementary.trigonometric.sin):
+        return sympy_pb2.Function.Sin
+    elif isinstance(expr, sympy.functions.elementary.trigonometric.cos):
+        return sympy_pb2.Function.Cos
+    elif isinstance(expr, sympy.functions.elementary.trigonometric.tan):
+        return sympy_pb2.Function.Tan
+    else:
+        return sympy_pb2.Function.NONE
+
+
+def _get_sympy_function_from_enum(enum: int) -> Any:
+    """
+    Helper function for sympy function deserialization.
+
+    Sympy functions are represented as a sympy_pb2.Function enum.  This mehtod converts this int enum
```

Comment:
```suggestion
    Sympy functions are represented as a sympy_pb2.Function enum.  This method converts this int enum
```

Reply:
Make sure the docstrings are at-most 100 characters wide as per the style guide.